### PR TITLE
Add detection rule for hp-socket linking

### PR DESCRIPTION
**Pull request by Still34**
## Summary

`hp-socket` is an open-source Chinese-origin networking library and has been spotted in various proprietary RAT such as [SprySocks](https://www.trendmicro.com/en_us/research/23/i/earth-lusca-employs-new-linux-backdoor.html)

### CODE REVIEW

**Review by mr-tz — APPROVED** [review]
thanks, once we have a test sample, this is good to go

**Review by mike-hunhoff — APPROVED** [review]
Awesome!

File: linking/static/hp-socket/linked-against-hp-socket.yml [hunk 1]
```diff
@@ -0,0 +1,30 @@
+rule:
+  meta:
+    name: linked against hp-socket
+    namespace: linking/static/hp-socket
+    authors:
+      - still@teamt5.org
+    scope: file
```
Comment (yelhamer) [inline linking/static/hp-socket/linked-against-hp-socket.yml:7]: capa now has initial support for dynamic analysis. So rules must provide their scopes in both static (file, function, basic block, instruction, or unsupported) and dynamic (file, process, thread, call, or unsupported) analysis flavors. Please check the updated [doc/format.md](https://github.com/mandiant/capa-rules/blob/master/doc/format.md#meta-block) for more info :]

Please consider the following suggestion:
```suggestion
    scopes:
      static: file
      dynamic: file
```

**Review by yelhamer — CHANGES_REQUESTED** [review]

**Review by yelhamer — APPROVED** [review]
Thanks!

**Comment by mr-tz** [comment]
I wasn't able to find a sample that matches AND is quick to analyze for testing. Can we move this to the nursery for now?

**Comment by Still34** [comment]
Is that something you'd like me to do? Move the file to the nursery section?

**Comment by mr-tz** [comment]
Yes, please, if you can.

**Comment by mr-tz** [comment]
thank you!

**Pull request merged**